client/dashboard/sites/settings-sftp-ssh/ssh-card.tsx

```diff
@@ -0,0 +1,304 @@
+		} );
+	};
+
+	const fields: Field< SshCardFormData >[] = [
+		{
+			id: 'connection_command',
+			label: __( 'Connection command' ),
+			Edit: ( { field, data } ) => {
+				return (
+					<ClipboardInputControl
+						label={ field.label }
+						value={ field.getValue( { item: data } ) }
+						readOnly
+						__next40pxDefaultSize
+						onCopy={ handleCopy }
+					/>
+				);
+			},
+		},
+		{
+			id: 'ssh_key',
+			label: __( 'SSH key' ),
+			Edit: ( { field, data } ) => {
+				const value = field.getValue( { item: data } );
+				if ( siteSshKeys && siteSshKeys.length > 0 ) {
+					return (
+						<BaseControl label={ field.label } __nextHasNoMarginBottom>
+							<VStack>
+								{ siteSshKeys.map( ( siteSshKey: SiteSshKey ) => (
+									<SshKeyCard
+										key={ siteSshKey.sha256 }
+										siteSshKey={ siteSshKey }
+										userLocale={ userLocale }
+										isBusy={ detachSshKeyMutation.isPending }
+										onDetach={ handleDetachSshKey }
+									/>
+								) ) }
+							</VStack>
+						</BaseControl>
+					);
+				}
+
+				// TODO: Add ReauthRequired
+				return (
+					<SelectControl
+						label={ field.label }
+						value={ value }
+						options={ field.elements ?? [] }
+						onChange={ ( value ) => onChange( { [ id ]: value } ) }
+						__next40pxDefaultSize
+						__nextHasNoMarginBottom
+					/>
+				);
+			},
+			elements: profileSshKeys?.map( ( profileSshKey: ProfileSshKey ) => ( {
+				label: `${ user.username }-${ profileSshKey.name }`,
+				value: profileSshKey.name,
+			} ) ),
+		},
+	];
+
+	const form = {
+		type: 'regular' as const,
+		fields: [ 'connection_command', 'ssh_key' ],
+	};
+
+	return (
+		<Card>
+			<CardBody>
+				<VStack style={ { paddingBottom: '12px' } }>
+					<Text size="15px" weight={ 500 } lineHeight="32px">
+						{ __( 'SSH' ) }
+					</Text>
+					<Text variant="muted" as="p">
+						{ createInterpolateElement(
+							__(
+								"SSH lets you access your site's backend via a terminal, so you can manage files and use <wpCliLink>WP-CLI</wpCliLink> for quick changes and troubleshooting. <learnMoreLink>Learn more</learnMoreLink>."
+							),
+							{
+								wpCliLink: <ExternalLink href="#" children={ null } />,
+								learnMoreLink: <ExternalLink href="#hosting-connect-to-ssh" children={ null } />,
+							}
+						) }
+					</Text>
+				</VStack>
+				<VStack spacing={ 4 } style={ { padding: '8px 0' } }>
+					<ToggleControl
+						label={ __( 'Enable SSH access for this site' ) }
+						checked={ sshEnabled }
+						disabled={ toggleSshAccessMutation.isPending }
+						onChange={ handleToggleSshAccess }
+						__nextHasNoMarginBottom
+					/>
+					{ sshEnabled && (
+						<DataForm< SshCardFormData >
+							data={ formData }
+							fields={ fields }
+							form={ form }
+							onChange={ ( edits: Partial< SshCardFormData > ) => {
+								setFormData( ( data ) => ( { ...data, ...edits } ) );
+							} }
+						/>
+					) }
+				</VStack>
+				{ showSshKeysSelect && (
```

Comment:
There's no way to add an SSH key if you don’t already have one attached to your account.

<img width="641" alt="Screenshot 2025-06-03 at 11 54 31" src="https://github.com/user-attachments/assets/9559059a-b46d-4251-9a39-7b773f4dc46b" />

We should disable the key select and attach button in this case, but maintain the button for users to add a new SSH key.

![image](https://github.com/user-attachments/assets/20de1ea9-e579-4fde-95d9-fba253074f82)

Reply:
Nice catch!